CMakeLists.txt

```diff
@@ -543,12 +543,10 @@ set_target_properties(torrent-rasterbar
 		SOVERSION ${SOVERSION}
 )
 
-if (cxx_std_14 IN_LIST CMAKE_CXX_COMPILE_FEATURES)
-	target_compile_features(torrent-rasterbar PUBLIC cxx_std_14)
-	message(STATUS "Building in C++14 mode")
-else()
-	target_compile_features(torrent-rasterbar PUBLIC cxx_std_11)
-	message(STATUS "Building in C++11 mode")
+if (NOT "${CXX_STANDARD}" STREQUAL "")
```

Comment:
The name of the variable is `CMAKE_CXX_STANDARD`, `CXX_STANDARD` is the name of the property.

Reply:
oh. I'm confused. I was seeing an error on appveyor (before I added the `-DCMAKE_CXX_VERSION` to the command line) where `_std` expanded into some weird "NOTFOUND" string. I thought this was a way to detect whether the property was set or not. I suppose it isn't.

It looks like the value "NOTFOUND" is well defined. So I could just check for that then, right?

Reply:
For a property that can have default value you set it via variable named "CMAKE_<property name>";  CMake copies values from such variables into properties when creating a target.

Reply:
> It looks like the value "NOTFOUND" is well defined. So I could just check for that then, right?

You can even write 
```cmake
if (NOT ${_std})
```
`NOTFOUND` is one of the boolean "false" values for CMake.

Reply:
Pardon, `(if ${_std})`, of course.

Reply:
> For a property that can have default value you set it via variable named CMAKE_<property name>; CMake copies values from such variables into properties when creating a target.

makes sense. so my understanding then is to never set any `CMAKE_` variables inside a CMakeList.txt, but only set them at the top level `cmake` invocation, otherwise you risk different targets to have different properties.

I'll simplify the check.